airgun/entities/contentview_new.py

```diff
@@ -22,6 +29,57 @@ def search(self, value):
         view = self.navigate_to(self, 'All')
         return view.search(value)
 
+    def publish(self, entity_name, values=None):
+        """Publishes to create new version of CV and promotes the contents to
+        'Library' environment.
+        :return: dict with new content view version table row; contains keys
+            like 'Version', 'Status', 'Environments' etc.
+        """
+        view = self.navigate_to(self, 'Publish', entity_name=entity_name)
+        if values:
+            view.fill(values)
+        view.next.click()
+        view.finish.click()
+        view.progressbar.wait_for_result()
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        return view.versions.table.read()
+
+    def promote(self, entity_name, version_name, lce_name):
+        """Promotes the selected version of content view to given environment.
+        :return: dict with new content view version table row; contains keys
+            like 'Version', 'Status', 'Environments' etc.
+        """
+        view = self.navigate_to(self, 'Promote', entity_name=entity_name, version_name=version_name)
+        modal = NewContentViewVersionPromote(self.browser)
+        if modal.is_displayed:
+            modal.lce.fill({lce_name: True})
+            modal.promote.click()
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.versions.search(version_name)
+        return view.versions.table.row(version=version_name).read()
+
+    def publish_and_promote(self, entity_name, lce_name, values=None):
+        view = self.navigate_to(self, 'Publish', entity_name=entity_name)
+        if values:
+            view.fill(values)
+        view.lce(lce_name).fill(True)
+        view.next.click()
+        view.next.click()
+        view.progressbar.wait_for_result()
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        return view.versions.table.read()
+
+    def update(self, entity_name, values):
+        """Update existing content view"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        # need a wait to recognize the loading is complete
+        # sleep works for now
+        time.sleep(3)
```

Comment:
no hard sleep!

Reply:
this was the old me omkar, I can fix it :)

Reply:
returning to this: have tried various other things, and the sleep is the only thing consistently working

Reply:
I would say keep it then if there is no other way. 
As I believe in many cases there is unfortunately none. :/

Reply:
And then I would remove `# sleep works for now` comment :)